### ISSUE TITLE
nixos-install fails with error: path '/nix/store/gy4yv67gv3j6in0lalw37j353zdmfcwm-nix-1.11.16' is not valid

### ISSUE THREAD

**Issue by jpotier** [NixOS/nixpkgs#33990]
## Issue description

`nixos-install` fails with `error: path '/nix/store/gy4yv67gv3j6in0lalw37j353zdmfcwm-nix-1.11.16' is not valid`

### Steps to reproduce

Follow the instructions there: https://nixos.wiki/wiki/Install_NixOS_on_Online.Net
to bootstrap nixos from Ubuntu 16.04.3 LTS (xenial) x86_64

Used to run without troubles. I suspect something changed with recent nix.

## Technical details

 - system: `"x86_64-linux"`
 - host os: `Linux 4.4.0-97-generic, Ubuntu, 16.04.3 LTS (Xenial Xerus)`
 - multi-user?: `no`
 - sandbox: `no`
 - version: `nix-env (Nix) 1.11.16`
 - channels(root): `"nixos-17.09.2754.66b4de79e38"`
 - channels(eeva): `"nixos-17.09.2754.66b4de79e38"`
 - nixpkgs: `/root/.nix-defexpr/channels/nixos`

**Comment by jpotier** [comment]
Reproduced it with Ubuntu 14.04 image.

`tree -a /mnt` (before):

```
/mnt/
|-- boot
|-- etc
|   `-- nixos
|       |-- configuration.nix
|       |-- hardware-configuration.nix
|       `-- users.nix
`-- home
```

`nixos-install`: http://lpaste.net/3012850413465501696

`tree -a /mnt` (after): https://framadrop.org/r/7CSnUlbUbG#7PCZDqOGYZVtUwoPsPxZdkE1Sx3DCphnRH09pMmfl+c=

`strace` of the `nixos-install`: https://framadrop.org/r/jO4TCuUDfM#mGO+o5Xi3Ngd4ACVTHpSxBo5P8H4ODyW8dX9qtO23DM=

**Comment by jpotier** [comment]
Could this be related to commit 77ce02201e1bc7facabb774d7e9996ab803145d2? (@lheckemann)

**Comment by lheckemann** [comment]
I don't think so. You could try checking out the git repo and reverting that commit and installing from there?

```
# apt install -y git
# git clone https://github.com/nixos/nixpkgs
# cd nixpkgs
# git revert 77ce02201e1bc7facabb774d7e9996ab803145d2 --no-edit
# export NIX_PATH=nixpkgs=$PWD:nixos=$PWD/nixos
```
Then proceed with the instructions, bar any setting of NIX_PATH.

I'll also see if I can reproduce this on another cloud host.

**Comment by lheckemann** [comment]
Update: I can reproduce this on a scaleway machine. Reverting 77ce022 did not help. I'll try bisecting.

**Comment by gbuisson** [comment]
Using the workaround, how did you manage to get enough free space on the rescue filesystem?
It's already very crowded there, I can't build the system if I also have to checkout a git repo...

**Comment by lheckemann** [comment]
You can clone it into the target filesystem.

**Comment by gbuisson** [comment]
I mounted it with a `mount -o bind` ;-) thanks

**Comment by jpotier** [comment]
Didn't to do anything special, just used `git clone --depth=1`

**Comment by gbuisson** [comment]
dumb question hopefully, but if I follow @lheckemann instructions, this command:

`nix-env -i -A config.system.build.nixos-install -A config.system.build.nixos-option -A config.system.build.nixos-generate-config -f "<nixos>"`

fails with `attribute �system� in selection path �config.system.build.nixos-install� not found`

I just cloned the repo, set the export, setup a mount to get additional space and run the instructions blindly step by step.

**Comment by lheckemann** [comment]
@gbuisson your NIX_PATH entry for nixos is incorrect. Did you do `export NIX_PATH=nixpkgs=$PWD:nixos=$PWD/nixos` as written in my instructions? Using `-f '<nixpkgs/nixos>'` instead of `-f '<nixos>' should also work.

**Comment by jpotier** [comment]
@gbuisson I admit there may be a hole in the instructions, since those were not assembled with this latest "bug" in mind. I can say that the installation ran fine with the `release-17.09` branch of nixpkgs from a few days ago.

**Comment by gbuisson** [comment]
@jpotier Thank you, actually my `NIX_PATH` was wrong because I set it with the command @lheckemann supplied but from another directory...

I also had to skip adding the nix channel (as marked on the instructions) and make some room binding a local folder on the target partition.

**Comment by jpotier** [comment]
@gbuisson yes exactly. So you managed to install your machine, right?

Let's hope this nix-channel thing is  only  transient.

**Comment by gbuisson** [comment]
Yes I did, I had some other unrelated issues but it's all fixed now thank
you  ;-)

**Comment by bendlas** [comment]
I'm getting this error as well, when trying to use nixos-install from a recent git checkout:
`error: path ‘/nix/store/xryr9g56h8yjddp89d6dw12anyb4ch7c-nix-1.10’ is not valid` 

I can reproduce this with `nixos-install --root /tmp/sys` with the config:
```nix
{ config, pkgs, ... }: {
  boot.isContainer = true;
  boot.loader.grub.enable = false;
  networking.hostId = "cafebabe";
  system.stateVersion = "18.03";
}
```
cc @copumpkin

**Comment by bendlas** [comment]
Pretty sure, the error message originates from https://github.com/NixOS/nix/blob/84722d67d2b09b5c28e1c2d9dd438ba592df2296/src/libstore/local-fs-store.cc#L83, so cc @edolstra as well
